### PR TITLE
fixed_math: add version 2.0.0, fix install path

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **fixed_math/***

#### Motivation
- fixed_math becomes header only
- add version 2.0.0
- fix install path (https://github.com/arturbac/fixed_math/issues/8)

#### Details
https://github.com/arturbac/fixed_math/compare/v1.0.2...v2.0.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by uilianries** [comment]
@toge Thank you for your PR :smile: I see it should add the version 2.0.0, but I only see the version 1.0.2 in this PR, is something missing?

**Comment by toge** [comment]
@uilianries 
It is strange.
I wrote 2.0.0 definition in https://github.com/conan-io/conan-center-index/pull/25269/files#diff-c646d0a6f9955b287a7d2328cb27c283c80e92e2181897f8f318746ce547e9c3R2

File: recipes/fixed_math/all/conanfile.py [hunk 1]
```diff
@@ -17,58 +15,30 @@ class FixedMathConan(ConanFile):
...
-            "gcc": "10", # fixed_math requires __has_builtin
-            "clang": "7",
-            "apple-clang": "12",
-            "Visual Studio": "16",
-            "msvc": "192",
+            "Visual Studio": "17",
+            "msvc": "193",
+            "gcc": "12",
+            "clang": "15",
+            "apple-clang": "14",
```
Comment (AbrilRBS) [inline recipes/fixed_math/all/conanfile.py:34]: I'm getting compilation errors in `apple-clang <=15`, seems like `apple-clang 16` is needed for this one.

in fact:
```
$ conan create . --version=2.0.0 -b=missing -s=compiler.cppstd=23 -s=compiler.version=16
Migration: Successfully updated settings.yml

======== Exporting recipe to the cache ========
fixed_math/2.0.0: Exporting package recipe: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/conanfile.py
fixed_math/2.0.0: exports: File 'conandata.yml' found. Exporting it...
fixed_math/2.0.0: Copied 1 '.py' file: conanfile.py
fixed_math/2.0.0: Copied 1 '.yml' file: conandata.yml
fixed_math/2.0.0: Exported to cache folder: /Users/abril/.conan2/p/fixed31e86af06d8b1/e
fixed_math/2.0.0: Exported: fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b (2024-09-16 15:48:22 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=23
compiler.libcxx=libc++
compiler.version=16
os=Macos
[platform_tool_requires]
cmake/3.29.6
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]


Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
[platform_tool_requires]
cmake/3.29.6
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]



======== Computing dependency graph ========
Graph root
    cli
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b - Cache

======== Computing necessary packages ========
fixed_math/2.0.0: Main binary package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' missing
fixed_math/2.0.0: Checking 0 compatible configurations
fixed_math/2.0.0: Compatible configurations not found in cache, checking servers
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Build

======== Installing packages ========
fixed_math/2.0.0: Calling source() in /Users/abril/.conan2/p/fixed31e86af06d8b1/s/src
fixed_math/2.0.0: Unzipping v2.0.0.tar.gz to .

-------- Installing package fixed_math/2.0.0 (1 of 1) --------
fixed_math/2.0.0: Building from source
fixed_math/2.0.0: Package fixed_math/2.0.0:da39a3ee5e6b4b0d3255bfef95601890afd80709
fixed_math/2.0.0: Building your package in /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/b
fixed_math/2.0.0: Generating aggregated env files
fixed_math/2.0.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
fixed_math/2.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' built
fixed_math/2.0.0: Build folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/b/build-release
fixed_math/2.0.0: Generating the package
fixed_math/2.0.0: Packaging in folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/p
fixed_math/2.0.0: Calling package()
fixed_math/2.0.0: package(): Packaged 1 file: LICENCE
fixed_math/2.0.0: package(): Packaged 10 '.h' files
fixed_math/2.0.0: package(): Packaged 1 '.hpp' file: fixed_math.hpp
fixed_math/2.0.0: Created package revision 0005566d694e40d43bc25465c573b4bb
fixed_math/2.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created
fixed_math/2.0.0: Full package reference: fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709#0005566d694e40d43bc25465c573b4bb
fixed_math/2.0.0: Package folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    fixed_math/2.0.0 (test package): /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/conanfile.py
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b - Cache

======== Computing necessary packages ========
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709#0005566d694e40d43bc25465c573b4bb - Cache

======== Installing packages ========
fixed_math/2.0.0: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Test package build: build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Test package build folder: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Writing generators to /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators
fixed_math/2.0.0 (test package): Generator 'CMakeDeps' calling 'generate()'
fixed_math/2.0.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(fixed_math)
    target_link_libraries(... fixed_math::fixed_math)
fixed_math/2.0.0 (test package): Generator 'CMakeToolchain' calling 'generate()'
fixed_math/2.0.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
fixed_math/2.0.0 (test package): CMakeToolchain generated: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators/CMakePresets.json
fixed_math/2.0.0 (test package): CMakeToolchain generated: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/CMakeUserPresets.json
fixed_math/2.0.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
fixed_math/2.0.0 (test package): Generating aggregated env files
fixed_math/2.0.0 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
fixed_math/2.0.0 (test package): Calling build()
fixed_math/2.0.0 (test package): Running CMake.configure()
fixed_math/2.0.0 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package"
-- Using Conan toolchain: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 23 with extensions OFF
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/abril/Downloads/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'fixed_math::fixed_math'
-- Configuring done (5.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release

fixed_math/2.0.0 (test package): Running CMake.build()
fixed_math/2.0.0 (test package): RUN: cmake --build "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
fixed_math/2.0.0 (test package): Running test()
fixed_math/2.0.0 (test package): RUN: ./test_package
31.9259 1.0101165771484375
```

**Review by AbrilRBS — COMMENTED** [review]
Thanks!
Successful compilation logs for apple-clang 16:

```
$ conan create . --version=2.0.0 -b=missing -s=compiler.cppstd=23 -s=compiler.version=16
Migration: Successfully updated settings.yml

======== Exporting recipe to the cache ========
fixed_math/2.0.0: Exporting package recipe: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/conanfile.py
fixed_math/2.0.0: exports: File 'conandata.yml' found. Exporting it...
fixed_math/2.0.0: Copied 1 '.py' file: conanfile.py
fixed_math/2.0.0: Copied 1 '.yml' file: conandata.yml
fixed_math/2.0.0: Exported to cache folder: /Users/abril/.conan2/p/fixed31e86af06d8b1/e
fixed_math/2.0.0: Exported: fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b (2024-09-16 15:48:22 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=23
compiler.libcxx=libc++
compiler.version=16
os=Macos
[platform_tool_requires]
cmake/3.29.6
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]


Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
[platform_tool_requires]
cmake/3.29.6
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]



======== Computing dependency graph ========
Graph root
    cli
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b - Cache

======== Computing necessary packages ========
fixed_math/2.0.0: Main binary package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' missing
fixed_math/2.0.0: Checking 0 compatible configurations
fixed_math/2.0.0: Compatible configurations not found in cache, checking servers
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Build

======== Installing packages ========
fixed_math/2.0.0: Calling source() in /Users/abril/.conan2/p/fixed31e86af06d8b1/s/src
fixed_math/2.0.0: Unzipping v2.0.0.tar.gz to .

-------- Installing package fixed_math/2.0.0 (1 of 1) --------
fixed_math/2.0.0: Building from source
fixed_math/2.0.0: Package fixed_math/2.0.0:da39a3ee5e6b4b0d3255bfef95601890afd80709
fixed_math/2.0.0: Building your package in /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/b
fixed_math/2.0.0: Generating aggregated env files
fixed_math/2.0.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
fixed_math/2.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' built
fixed_math/2.0.0: Build folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/b/build-release
fixed_math/2.0.0: Generating the package
fixed_math/2.0.0: Packaging in folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/p
fixed_math/2.0.0: Calling package()
fixed_math/2.0.0: package(): Packaged 1 file: LICENCE
fixed_math/2.0.0: package(): Packaged 10 '.h' files
fixed_math/2.0.0: package(): Packaged 1 '.hpp' file: fixed_math.hpp
fixed_math/2.0.0: Created package revision 0005566d694e40d43bc25465c573b4bb
fixed_math/2.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created
fixed_math/2.0.0: Full package reference: fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709#0005566d694e40d43bc25465c573b4bb
fixed_math/2.0.0: Package folder /Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    fixed_math/2.0.0 (test package): /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/conanfile.py
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b - Cache

======== Computing necessary packages ========
Requirements
    fixed_math/2.0.0#623ed3f319c4a6e53166796d75baf98b:da39a3ee5e6b4b0d3255bfef95601890afd80709#0005566d694e40d43bc25465c573b4bb - Cache

======== Installing packages ========
fixed_math/2.0.0: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Test package build: build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Test package build folder: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release
fixed_math/2.0.0 (test package): Writing generators to /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators
fixed_math/2.0.0 (test package): Generator 'CMakeDeps' calling 'generate()'
fixed_math/2.0.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(fixed_math)
    target_link_libraries(... fixed_math::fixed_math)
fixed_math/2.0.0 (test package): Generator 'CMakeToolchain' calling 'generate()'
fixed_math/2.0.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
fixed_math/2.0.0 (test package): CMakeToolchain generated: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators/CMakePresets.json
fixed_math/2.0.0 (test package): CMakeToolchain generated: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/CMakeUserPresets.json
fixed_math/2.0.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
fixed_math/2.0.0 (test package): Generating aggregated env files
fixed_math/2.0.0 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
fixed_math/2.0.0 (test package): Calling build()
fixed_math/2.0.0 (test package): Running CMake.configure()
fixed_math/2.0.0 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package"
-- Using Conan toolchain: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 23 with extensions OFF
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/abril/Downloads/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'fixed_math::fixed_math'
-- Configuring done (5.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release

fixed_math/2.0.0 (test package): Running CMake.build()
fixed_math/2.0.0 (test package): RUN: cmake --build "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-16-armv8-23-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
fixed_math/2.0.0 (test package): Running test()
fixed_math/2.0.0 (test package): RUN: ./test_package
31.9259 1.0101165771484375
```

File: recipes/fixed_math/all/conanfile.py (hunk 1 above)
Comment (AbrilRBS) [inline recipes/fixed_math/all/conanfile.py:34]:
```suggestion
            "apple-clang": "16",
```

Compilation log errors for apple-clang 15:

```
======== Testing the package: Building ========
fixed_math/2.0.0 (test package): Calling build()
fixed_math/2.0.0 (test package): Running CMake.configure()
fixed_math/2.0.0 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package"
-- Using Conan toolchain: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-15-armv8-23-release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 23 with extensions OFF
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'fixed_math::fixed_math'
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-15-armv8-23-release

fixed_math/2.0.0 (test package): Running CMake.build()
fixed_math/2.0.0 (test package): RUN: cmake --build "/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/build/apple-clang-15-armv8-23-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/test_package.cpp:10:26: error: constexpr variable 'foo_constant' must be initialized by a constant expression
inline constexpr fixed_t foo_constant{fixedmath::fobj::tan(15 * fixedmath::phi / 180)};
                         ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/abril/.conan2/p/b/fixed8b9b2a9f357c4/p/include/fixedmath/math.h:656:26: note: control flows through the definition of a static variable
    static constexpr int prec_ = 16;
                         ^
/Users/abril/coding/conan-center-index/recipes/fixed_math/all/test_package/test_package.cpp:10:39: note: in call to '&tan->operator()({17157})'
inline constexpr fixed_t foo_constant{fixedmath::fobj::tan(15 * fixedmath::phi / 180)};
                                      ^
1 error generated.
make[2]: *** [CMakeFiles/test_package.dir/test_package.cpp.o] Error 1
make[1]: *** [CMakeFiles/test_package.dir/all] Error 2
make: *** [all] Error 2

ERROR: fixed_math/2.0.0 (test package): Error in build() method, line 21
	cmake.build()
	ConanException: Error 2 while executing
```

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Comment by toge** [comment]
@AbrilRBS 
Thank you for your improvement!

**Review by danimtb — APPROVED** [review]

**Pull request merged**